msdsales/params.go

```diff
@@ -0,0 +1,69 @@
+package msdsales
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/paramsbuilder"
+	"golang.org/x/oauth2"
+)
+
+const (
+	// MaxPageSize is upper limit on elements per page.
+	MaxPageSize = 100
+)
+
+// Option is a function which mutates the connector configuration.
+type Option func(params *parameters)
+
+// parameters MS Sales supports auth client, workspace, etc. by delegation.
+type parameters struct {
+	paramsbuilder.Client
+	paramsbuilder.Workspace
+	paramsbuilder.Module
```

Comment:
MS Sales allows configuration via Options Pattern just like other connectors.
Other connectors would compose paramteres like this by delegation.
Reason: If underlying Client parameter changes it will be applied to all parameters that delegate to it.

Reply:
https://www.loom.com/share/97b52ea5eeab449cb0dd00abc2842785